### PR TITLE
Update docs for api endpoints

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

Add docs for undocumented api endpoints.
.

### CODE REVIEW

File: book/src/http_advanced.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Lighthouse REST API: `/spec`
+
+The `/advanced` endpoints provide information Lighthouse specific data structures for advanced debugging.
+
+## Endpoints
+
+HTTP Path | Description |
+| --- | -- |
+[`/advanced/fork_choice`](#advancedfork_choice) | Get the `proto_array` fork choice object.
+[`/spec/slots_per_epoch`](#advancedoperation_pool) | Get the Lighthouse `PersistedOperationPool` object.
```
Comment (paulhauner) [inline book/src/http_advanced.md]: `/spec/slots_per_epoch` should be something else :)

File: book/src/http_beacon.md [hunk 2]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/state/genesis`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200, 404
+
+### Parameters
```
Comment (paulhauner) [inline book/src/http_beacon.md]: Maybe we should say "No Parameters" or something like that? I'm down with whatever is consistent with what we already have.

File: book/src/http_beacon.md [hunk 3]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/genesis_time`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200, 404
+
+### Parameters
```
Comment (paulhauner) [inline book/src/http_beacon.md]: As above

File: book/src/http_beacon.md [hunk 4]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/state/genesis`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200, 404
```
Comment (paulhauner) [inline book/src/http_beacon.md]: We can remove `404`, I don't think that's possible.

File: book/src/http_beacon.md [hunk 5]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/genesis_time`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200, 404
```
Comment (paulhauner) [inline book/src/http_beacon.md]: We can also remove `404`

File: book/src/http_beacon.md [hunk 6]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/fork`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200, 404
```
Comment (paulhauner) [inline book/src/http_beacon.md]: Can also remove `404`

File: book/src/http_beacon.md [hunk 7]
```diff
@@ -233,6 +236,105 @@ and its tree hash root.
...
+
+### Parameters
```
Comment (paulhauner) [inline book/src/http_beacon.md]: As above

File: book/src/http_beacon.md (hunk 2 above)
Comment (pawanjay176) [reply book/src/http_beacon.md]: I guess simply removing the `Parameters` heading when `Query Parameters` is the best choice and consistent with what we already have.

**Pull request merged**